lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -0,0 +1,209 @@
+			stagingDir,
+		});
+		this._loadDistMetadata = null;
+		this._sources = !!sources;
+	}
+	loadDistMetadata() {
+		if (!this._loadDistMetadata) {
+			this._loadDistMetadata = Promise.resolve().then(async () => {
+				const version = this._version;
+				log.verbose(
+					`Installing ${DIST_ARTIFACT_ID} in version ${version}...`
+				);
+
+				const {pkgPath: distPkgPath} = await this._installer.installPackage({
+					pkgName: DIST_PKG_NAME,
+					groupId: DIST_GROUP_ID,
+					artifactId: DIST_ARTIFACT_ID,
+					version,
+					classifier: "npm-sources",
+					extension: "zip",
+				});
+
+				return await this._installer.readJson(
+					path.join(distPkgPath, "metadata.json")
+				);
+			});
+		}
+		return this._loadDistMetadata;
+	}
+	async getLibraryMetadata(libraryName) {
+		const distMetadata = await this.loadDistMetadata();
+		const metadata = distMetadata.libraries[libraryName];
+
+		if (!metadata) {
+			throw new Error(`Could not find library "${libraryName}"`);
+		}
+
+		return metadata;
+	}
+	async handleLibrary(libraryName) {
+		const metadata = await this.getLibraryMetadata(libraryName);
+		if (!metadata.gav) {
+			throw new Error(
+				"Metadata is missing GAV (group, artifact and version) " +
+					"information. This might indicate an unsupported SNAPSHOT version."
+			);
+		}
+		const gav = metadata.gav.split(":");
+		let pkgName = metadata.npmPackageName;
+		if (!this._sources) {
+			pkgName += "-prebuilt";
+		}
+
+		return {
+			metadata: Promise.resolve({
+				id: pkgName,
+				version: metadata.version,
+				dependencies: metadata.dependencies,
+				optionalDependencies: metadata.optionalDependencies,
+			}),
+			// Trigger installation of package
+			install: this._installer.installPackage({
+				pkgName,
+				groupId: gav[0],
+				artifactId: gav[1],
+				version: metadata.version,
+				classifier: this._sources ? "npm-sources" : null,
+				extension: this._sources ? "zip" : "jar",
+			}),
+		};
+	}
+	static async fetchAllVersions({ui5HomeDir, cwd} = {}) {
+		const installer = new Installer({
+			cwd: cwd ? path.resolve(cwd) : process.cwd(),
+			ui5HomeDir: path.resolve(
+				ui5HomeDir || path.join(os.homedir(), ".ui5")
+			),
+			snapshotEndpointUrlCb: this._snapshotEndpointUrlCb,
+		});
+		return await installer.fetchPackageVersions({
+			groupId: DIST_GROUP_ID,
+			artifactId: DIST_ARTIFACT_ID,
+		});
+	}
+	/**
+	 * Tries to detect whether ~/.m2/settings.xml exist, and if so, whether
+	 * the snapshot.build URL is extracted from there
+	 *
+	 * @param {string} [settingsXML=~/.m2/settings.xml] Path to the settings.xml.
+	 * 				If not provided, the default location is used
+	 * @param {boolean} [skipConfirmation=false] Whether to skip manual user
+	 * 				confirmation in the consle and enforce reading from settings.xml
+	 * @returns {Promise<string>} The resolved snapshot.build URL from ~/.m2/settings.xml
+	 */
+	static async _resolveSnapshotEndpointUrl(settingsXML, skipConfirmation) {
+		if (!skipConfirmation) {
```

Comment:
Maybe this has already been discussed, but I'm wondering whether we should first attempt to read the `settings.xml` file, and then prompt the user whether to use the content.

In case a user does not have a settings.xml file, we would prompt them whether to use the file, only to then fail with an error because the file does not exist. I think we should avoid this.